File: src/CSharp/748.ShortestCompletingWord.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace PeefyLeetCode.ShortestCompletingWord
{
public class Solution {
    public string ShortestCompletingWord(string licensePlate, string[] words) {
        licensePlate = licensePlate.ToLower();
        int[] cnt = new int[26];
        foreach (char c in licensePlate){
            if(c >='a' && c <= 'z') cnt[c - 'a']++;
        }
        string ans = "";
        foreach (string w in words){
            int[] cnt2 = new int[26];
            foreach (char c in w){
                if(c>='a' && c<='z') cnt2[c - 'a']++;
            }
            int p = 0;
            while(p < 26){
                if(cnt2[p] < cnt[p]) break;
                p++;
            }
            if(p == 26){
                if(ans == "" || ans.Length > w.Length) 
                    ans = w;
            }
        }
        return ans;
    }
}
}

File: src/Java/701-800/748.ShortestCompletingWord.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int stringCount(String s, char c){
        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == c){
                count++;
            }
        }
        return count;
    }

    public String shortestCompletingWord(String licensePlate, String[] words) {
        HashMap<Character, Integer> hash = new HashMap<>();
        licensePlate = licensePlate.toLowerCase();
        for (char c : licensePlate.toCharArray()) {
            if ('a' <= c && c <= 'z')
                hash.put(c, hash.getOrDefault(c, 0) + 1);
        }
        int minlen = 16;
        String ret ="";
        for (String w : words){  
            if (w.length() >= minlen) {
                continue;
            }
            boolean candidate = true;
            for (char c : hash.keySet()) {
                if (stringCount(w, c) < hash.get(c)) {
                    candidate = false;
                    break;
                }
            }
            if (candidate == true){
                ret = w;
                minlen = w.length();
            }
        }
        return ret;
    }
}
